openedx/core/djangoapps/catalog/management/commands/cache_programs.py

```diff
@@ -43,32 +45,51 @@ def handle(self, *args, **options):
             raise
 
         programs = {}
+        pathways = {}
         for site in Site.objects.all():
             site_config = getattr(site, 'configuration', None)
             if site_config is None or not site_config.get_value('COURSE_CATALOG_API_URL'):
                 logger.info('Skipping site {domain}. No configuration.'.format(domain=site.domain))
                 cache.set(SITE_PROGRAM_UUIDS_CACHE_KEY_TPL.format(domain=site.domain), [], None)
+                cache.set(SITE_CREDIT_PATHWAY_IDS_CACHE_KEY_TPL.format(domain=site.domain), [], None)
                 continue
 
             client = create_catalog_api_client(user, site=site)
             uuids, program_uuids_failed = self.get_site_program_uuids(client, site)
             new_programs, program_details_failed = self.fetch_program_details(client, uuids)
+            new_pathways, pathways_failed = self.get_pathways(client, site)
+            new_pathways, new_programs, pathway_processing_failed = self.process_pathways(site, new_pathways,
```

Comment:
In this case, I think it may be cleaner to leave it as be, since I would still like the other items (other than the failure boolean) returned from these functions even if they fail. If get_pathways failed, I would need to add code to pass in an empty list to process_pathways(). If process_pathways() fails, I would still like the pathways and programs it was successfully able to process (I’ve updated to move the try/except to inside of the for block).